### PR TITLE
Don't pluralize with apostrophe s, plus a few other apostrophe fixes

### PR DESCRIPTION
**Pull request by n8sh**

### CODE REVIEW

File: articles/d-array-article.dd [hunk 1]
```diff
@@ -148,21 +148,21 @@ $(H2 $(LNAME2 determinism, Determinism))
 
 $(P There is one caveat with slice appending that can bite inexperienced, and even experienced D coders: the apparent non-deterministic behavior of appending.)
 
-$(P Let's say we have a function which is passed a buffer, and writes some number of A's to the buffer (appending if necessary), returning the filled buffer:)
+$(P Let's say we have a function which is passed a buffer, and writes some number of 'A's to the buffer (appending if necessary), returning the filled buffer:)
```
Comment (n8sh) [inline articles/d-array-article.dd]: Here we can add a leading apostrophe to make it correct because we're talking about the character `'A'`.

**Comment by thewilsonator** [comment]
Please remove the `function.dd` change into a separate PR because that will most probably cause this to fail.

**Comment by n8sh** [comment]
That's strange. Why will it (probably) do that?

**Comment by thewilsonator** [comment]
Because its too big and causes the doc generator to spew LaTeX errors. e.g. #2470 http://dtest.dlang.io/results/7d7cc2e5e4c8abac0eb5e76920efb05821fbe7e9/af558942e752f686d819f2a7eb1e54ee6b603e81/build.log

File: spec/function.dd [hunk 2]
```diff
@@ -1231,7 +1231,7 @@ int foo(in int x, out int y, ref int z, int q);
     $(TROW $(I none), parameter becomes a mutable copy of its argument)
 
     $(TROW $(D in), defined as $(D scope const).  However $(D in) has not yet been properly
-    implemented so it's current implementation is equivalent to $(D const).  It is recommended
+    implemented so its current implementation is equivalent to $(D const).  It is recommended
```
Comment (n8sh) [inline spec/function.dd]: Removing this from the PR as @thewilsonator suggested.

**Pull request merged**